src/main/java/org/trigon/hardware/phoenix6/talonfx/TalonFXMotor.java

```diff
@@ -29,68 +48,147 @@ public TalonFXMotor(int id, String motorName, String canbus) {
     }
 
+    /**
+     * Applies the configuration to be used both in real life and in simulation.
+     *
+     * @param simulationAndRealConfiguration the configuration
+     */
     public void applyConfiguration(TalonFXConfiguration simulationAndRealConfiguration) {
         motorIO.applyConfiguration(simulationAndRealConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used when {@link RobotHardwareStats#isSimulation()} is false.
+     * Having two different configurations allows for tuning motor behavior in simulation which might not perfectly mimic real life performance.
+     *
+     * @param realConfiguration the configuration
+     */
     public void applyRealConfiguration(TalonFXConfiguration realConfiguration) {
         if (!RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(realConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used in simulation.
+     * Having two different configurations allows for tuning motor behavior in simulation which might not perfectly mimic real life performance.
+     *
+     * @param simulationConfiguration the configuration
+     */
     public void applySimulationConfiguration(TalonFXConfiguration simulationConfiguration) {
         if (RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(simulationConfiguration);
     }
 
+    /**
+     * Stops the motor.
+     */
     public void stopMotor() {
         motorIO.stopMotor();
     }
 
+    /**
+     * Gets a signal from the motor.
+     *
+     * @param signal the type of signal to get
+     * @return the signal
+     */
     public double getSignal(TalonFXSignal signal) {
         return motorInputs.getSignal(signal.name);
     }
 
+    /**
+     * Gets a threaded signal from the motor.
+     * Threaded signals use threading to process certain signals separately. This is used for signals that need to be updated at a higher frequency such as odometry.
+     *
+     * @param signal the type of threaded signal to get
+     * @return the signal
+     */
     public double[] getThreadedSignal(TalonFXSignal signal) {
         return motorInputs.getThreadedSignal(signal.name);
     }
 
+    /**
+     * Registers a signal to the motor.
+     *
+     * @param signal               the signal to register
+     * @param updateFrequencyHertz the frequency at which the signal will be updated
+     */
     public void registerSignal(TalonFXSignal signal, double updateFrequencyHertz) {
         motorInputs.registerSignal(motorSignalToStatusSignal(signal), updateFrequencyHertz);
     }
 
+    /**
+     * Registers a threaded signal to the motor.
+     * Threaded signals use threading to process certain signals separately, keeping them running smoothly in the background.
+     * This avoids slowing down the main program, unlike simpler signals that run directly in it.
+     * This is also used for signals that need to be updated at a high frequency like for odometry.
```

Comment:
you're focusing too much on the unrelated stuff
the main reason we do this is for high freq signals
keep this jdoc consistent across all threaded stuff